(MODULES-4370) Fix CI for Windows 2008

jpogran opened the pull request.

jpogran committed:
(MODULES-4370) Fix CI for Windows 2008

glennsarti commented on spec/support/utilities/iis_application_pool.rb:
```diff
@@ -1,15 +1,43 @@
-  !(on(default, "C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe -noninteractive -noprofile -executionpolicy bypass -command \"& {Get-WebAppPoolState -Name #{@pool_name}}\"").stdout =~ /Started/i).nil?
+  command = []
+  command << "C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe -noninteractive -noprofile -executionpolicy bypass -command"
+  command << "\"& {"
```
Needs a space between command and `&`

glennsarti commented on spec/support/utilities/iis_application_pool.rb:
```diff
@@ -1,15 +1,43 @@
 def has_app_pool(pool_name)
-  !(on(default, "C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe -noninteractive -noprofile -executionpolicy bypass -command \"& {Get-WebAppPoolState -Name #{@pool_name}}\"").stdout =~ /Started/i).nil?
+  command = []
+  command << "C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe -noninteractive -noprofile -executionpolicy bypass -command"
```
Running old powershell via Cygwin has an annoying bug to with STDIN redirection

https://tickets.puppetlabs.com/browse/PE-18351?focusedCommentId=394366&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-394366

Fix was implemented here;
https://github.com/puppetlabs/pe_acceptance_tests/pull/1195

Simple as adding ` < /dev/null` to the end.

glennsarti commented:
@jpogran Can we put these changes under MODULES-4370 instead of maint?

jpogran commented:
Did not know MODULES-4370 existed. Will rename PR and modify commit message. This is also turning into a larger fix than a few simple edits so will split out commits.

glennsarti committed:
(MODULES-4370) Disable tests on Server 2008 due to OS differences

This commit disables various acceptance tests due to behvaioural differences
between Server 2008 and 2012.

- Get-Website does not honor the -Name parameter (MODULES-4463)
- APPCMD behaves erratically (MODULES-4464)

glennsarti committed:
(maint) Do not test for idle_time_action on IIS < 8.5

Previously the acceptance tests ran expectations on the idle_time_action
parameter, however this was only introduced in IIS 8.5.  This commit guards the
test to exclude the lower IIS versions.

glennsarti commented:
Merged without successful Travis CI run due to Travis having errors due to S3 outage.  Also ran locally and via adhoc pipelines.
